data-model/src/main/scala/za/co/absa/enceladus/model/Validation.scala

```diff
@@ -27,24 +27,42 @@ object Validation {
 
 }
 
-case class Validation (errors: Map[String, List[String]] = Map()) {
+case class Validation (errors: Map[String, List[String]] = Map(), warnings: Map[String, List[String]] = Map()) {
```

Comment:
I keep wondering if a different structure like the following would be better or worse:

```scala
  sealed trait IssueType
  object IssueTypes {
    case object Error extends IssueType
    case object Warning extends IssueType
  }
  
  case class Issue(tpe: IssueType, messages: List[String])

  case class Validation(issues: Map[String, Issue] = Map()) {...}
```

Some things could get simplified (like merging validations, 1-1 key-issueType mapping existence), some would get a tad harder (`def isValid = issues.values.exists(_.tpe == IssueTypes.Error)`).

Just thinking out loud, feel free to decide either way.

Reply:
After an offline discussion we came to a conclusion that we will keep the case class as is.